llvm/lib/CodeGen/AsmPrinter/DwarfDebug.h

```diff
@@ -158,7 +153,7 @@ class Multi {
 };
 /// Single location defined by (potentially multiple) MMI entries.
 struct MMI {
-  mutable SmallVector<FrameIndexExpr, 1> FrameIndexExprs;
+  std::set<FrameIndexExpr> FrameIndexExprs;
```

Comment:
Another possibilty would be a `SetVector`? (though then it's insertion ordered, rather than `<` ordered - not sure if that's sufficiently stable for the needs? perhaps not)

Reply:
I think we don't need the extra functionality provided by the `SetVector` here: our sorting keys (fragment offsets) are stable without the need for insertion-order-tracking.

Reply:
IIUC the DWARF emitter also wants the fragments ordered by offset? I'm not sure what would happen if we were to emit DWARF fragment expressions out of order 🤔

Reply:
> IIUC the DWARF emitter also wants the fragments ordered by offset? I'm not sure what would happen if we were to emit DWARF fragment expressions out of order 🤔

Ah, yeah - DWARF does require the fragments be in order (essentially they don't say where they are in the structure - so you have to emit the first byte first, second byte second, etc)